plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3ClientSettings.java

```diff
@@ -127,9 +131,13 @@
     /** Whether the s3 client should use an exponential backoff retry policy. */
     final boolean throttleRetries;
 
+    /** Whether the s3 client should use path style access. */
+    final boolean pathStyleAccess;
```

Comment:
The AWS SDK uses this as a ternary setting (true, false, null), with null being the default, which enables the path-style setting based on the configured endpoint. I think we should do treat this setting exactly the same way, allowing path-style setting both being explicitly enabled and disabled.

Reply:
Sounds good, so we break behavior after all by changing the default to `null`? See my comment above, this may not be necessary (at least in 7.x)?

Reply:
It's unclear what the behaviour of `setPathStyleAccessEnabled(Boolean.FALSE)` is from the SDK docs. I think we're doing the right thing here, either enabling it or leaving it unset.